src/ocean_emulators/utils/data.py

```diff
@@ -228,32 +230,18 @@ def from_dataset(cls, dataset: xr.Dataset, *, name: str = "dataset") -> Self:
     @classmethod
     def from_config(cls, cfg: TrainConfig | EvalConfig, *, use_dask: bool) -> Self:
         chunks: dict[str, int] | None = {} if use_dask else None
-
         root = cfg.experiment.data_dir
 
-        if "*" in cfg.data.data_path:
-            data = xr.open_dataset(
-                root / cfg.data.data_path,
-                engine="netcdf4",
-                chunks={"time": 1},
-            )
-        else:
-            data = xr.open_dataset(
-                root / cfg.data.data_path,
-                chunks=chunks,
-                consolidated=True,
-                engine="zarr",
-            )
-
-        means = xr.open_dataset(
-            root / cfg.data.data_means_path,
-            engine="netcdf4" if cfg.data.data_means_path.endswith(".nc") else "zarr",
-            chunks=chunks,
+        data = _open_dataset(cfg.data.data_path, root, chunks)
```

Comment:
🐑 🐑 🐑 
An idea to further reduce repetition: 
```suggestion
        open_ = functools.partial(_open_dataset, root=root, chunks=chunks)
        data, means, stds = open_(cfg.data.data_path), open_(cfg.data.data_means_path), open_(cfg.data.data_stds_path)
```

Reply:
Reading the `_open_dataset` function below, I almost think it would be better suited to be a function that addresses the data location, like `parse_location`, since the chunks and netcdf/zarr logic is otherwise the same.

Reply:
Combined with `functools.partial`, I think this could make for similarly succinct open call, while addressing the data location annoyance.

Reply:
I agree that we should break out the location logic, which we can do as discussed last week into another type. I try to avoid `functools.partial` because it defeats type checkers and in this case saves only minor and local duplication.